src/index.ts

```diff
@@ -1,84 +1,77 @@
-import * as path from 'path';
-import { Language } from 'language-recognizer/lib/types';
+import * as core from "@actions/core";
+import * as io from "@actions/io";
+import * as path from "path";
+import { BuildahCli, BuildahConfigSettings } from "./buildah";
 
 export async function run(): Promise<void> {
-
-    if (process.env.RUNNER_OS !== 'Linux') {
-        throw new Error('buildah, and therefore this action, only works on Linux. Please use a Linux runner.');
+    if (process.env.RUNNER_OS !== "Linux") {
+        throw new Error("buildah, and therefore this action, only works on Linux. Please use a Linux runner.");
     }
 
     // get buildah cli
-    const buildahPath = await io.which('buildah', true);
+    const buildahPath = await io.which("buildah", true);
     const cli: BuildahCli = new BuildahCli(buildahPath);
 
-    const workspace = process.env['GITHUB_WORKSPACE'];
-    let dockerFiles = getInputList('dockerfiles');
-    const newImage = `${core.getInput('image', { required: true })}:${core.getInput('tag', { required: true })}`;
-
-    const useOCI = core.getInput("oci") == "true";
+    const workspace = process.env.GITHUB_WORKSPACE || "";
+    const dockerFiles = getInputList("dockerfiles");
+    const image = core.getInput("image", { required: true });
+    const tags = core.getInput("tags");
+    const tagsList: string[] = tags.split(" ");
+    const newImage = `${image}:${tagsList[0]}`;
+    const useOCI = core.getInput("oci") === "true";
 
     if (dockerFiles.length !== 0) {
         await doBuildUsingDockerFiles(cli, newImage, workspace, dockerFiles, useOCI);
-    } else {
-        await doBuildFromScratch(cli, newImage, workspace, useOCI);
+    }
+    else {
+        await doBuildFromScratch(cli, newImage, useOCI);
     }
 
-    core.setOutput("image", newImage);
+    if (tagsList.length > 1) {
+        await cli.tag(image, tagsList);
+    }
+    core.setOutput("image", image);
+    core.setOutput("tags", tags);
 }
 
-async function doBuildUsingDockerFiles(cli: BuildahCli, newImage: string, workspace: string, dockerFiles: string[], useOCI: boolean): Promise<void> {
+async function doBuildUsingDockerFiles(
+    cli: BuildahCli, newImage: string, workspace: string, dockerFiles: string[], useOCI: boolean,
+): Promise<void> {
     if (dockerFiles.length === 1) {
         core.info(`Performing build from Dockerfile`);
     }
     else {
         core.info(`Performing build from ${dockerFiles.length} Dockerfiles`);
     }
 
-    const context = path.join(workspace, core.getInput('context'));
-    const buildArgs = getInputList('build-args');
-    dockerFiles = dockerFiles.map(file => path.join(workspace, file));
-    await cli.buildUsingDocker(newImage, context, dockerFiles, buildArgs, useOCI);
+    const context = path.join(workspace, core.getInput("context"));
+    const buildArgs = getInputList("build-args");
+    const inputdockerFiles = dockerFiles.map((file) => path.join(workspace, file));
```

Comment:
Since the `map` call here is joining the workspace path to the filename to create an absolute path, a better name for this variable would be `dockerFilePaths`.